### ISSUE TITLE
[Task]: Model Contract - PostProcessor & Preprocessor — Numpy Rewrite

### ISSUE THREAD

**Issue by rajeshgangireddy** [open-edge-platform/geti-instant-learn#1045]
### Description
Replace all `nn.Module`-based post-processors with pure numpy ABCs. Add `Preprocessor` ABC and pipeline.

#### 1. Base ABCs (`library/src/instantlearn/components/postprocessing/base.py`) — REPLACE

```python
### TBD: Does PreProcessor take Sample or just a numpy image ? 

class Preprocessor(ABC):
    @abstractmethod
    def __call__(self, image: np.ndarray) -> np.ndarray: ...

class PreprocessorPipeline(Preprocessor):
    def __init__(self, steps: list[Preprocessor]): ...
    def __call__(self, image: np.ndarray) -> np.ndarray: ...

class PostProcessor(ABC):
    @abstractmethod
    def __call__(self, masks: np.ndarray, scores: np.ndarray, labels: np.ndarray) -> tuple[np.ndarray, np.ndarray, np.ndarray]: ...

class PostProcessorPipeline(PostProcessor):
    def __init__(self, steps: list[PostProcessor]): ...
    def __call__(self, masks, scores, labels) -> tuple: ...
```

No `nn.Module`. No `forward()`. Plain `__call__()`.  
Pipeline operates on raw numpy arrays, not `Prediction` objects.

#### 2. NMS variants (`library/src/instantlearn/components/postprocessing/nms.py`) — REPLACE

All 5 NMS classes rewritten in pure numpy:

- `BoxNMS` — argsort scores, compute IoU via numpy, greedy suppression (replaces `torchvision.ops.nms`)
- `MaskNMS` — mask-IoU via boolean ops
- `BoxIoMNMS` — box intersection-over-minimum
- `MaskIoMNMS` — mask intersection-over-minimum (best for nested objects)
- `SoftNMS` — Gaussian score decay

#### 3. Filtering (`library/src/instantlearn/components/postprocessing/filtering.py`) — REPLACE

- `ScoreFilter(threshold: float)` — numpy boolean indexing
- `MinimumAreaFilter(min_area: int)` — sum mask pixels, filter

#### 4. Morphology (`library/src/instantlearn/components/postprocessing/morphology.py`) — REPLACE

- `MorphologicalOpening(kernel_size: int)` — use `cv2.morphologyEx(MORPH_OPEN)` on each mask
- `MorphologicalClosing(kernel_size: int)` — use `cv2.morphologyEx(MORPH_CLOSE)` on each mask

#### 5. Merge (`library/src/instantlearn/components/postprocessing/merge.py`) — REPLACE

- `MergePerClassMasks` — numpy `logical_or.reduce` per unique label

#### 6. Defaults (`library/src/instantlearn/components/postprocessing/defaults.py`) — REPLACE

- `default_postprocessor() -> PostProcessorPipeline` returns `PostProcessorPipeline([MaskIoMNMS(0.3), BoxIoMNMS(0.3)])`

### Files to modify/replace

| File | Action |
|------|--------|
| `library/src/instantlearn/components/postprocessing/base.py` | Replace |
| `library/src/instantlearn/components/postprocessing/nms.py` | Replace |
| `library/src/instantlearn/components/postprocessing/filtering.py` | Replace |
| `library/src/instantlearn/components/postprocessing/morphology.py` | Replace |
| `library/src/instantlearn/components/postprocessing/merge.py` | Replace |
| `library/src/instantlearn/components/postprocessing/defaults.py` | Replace |
| `library/src/instantlearn/components/postprocessing/__init__.py` | Update |

### Key design notes

- `apply_postprocessing()` helper removed — models call pipeline directly on raw arrays, then construct `Prediction`
- `BoxNMS` numpy implementation: sort by score descending, iterate, compute IoU with remaining, suppress above threshold
- Morphology: `cv2` is already a dependency; prefer `cv2.morphologyEx` over scipy
- All processors are `__call__`-based (not `forward`)

### Acceptance criteria

- [ ] All files pass `ruff check`
- [ ] Zero torch/torchvision imports in any postprocessing file
- [ ] `from instantlearn.components.postprocessing import PostProcessorPipeline, MaskIoMNMS` works without torch
- [ ] Existing NMS logic preserved (same thresholds, same algorithm, just numpy)

### Notes

- Model-specific ONNX-traceable processors (e.g., `Sam3Preprocessor`, `Sam3Postprocessor` in `models/sam3/processing.py`) stay as `nn.Module` — they're internal to the model and needed for export tracing. This issue only covers the shared pipeline.
